internal/usecase/client/pendo/pendo_client.go

```diff
@@ -0,0 +1,205 @@
+	logger := app_context.LogFromCtx(ctx)
+	if err := c.guardSendTrackEvent(track); err != nil {
+		logger.Error("wrong arguments: " + err.Error())
+		return err
+	}
+
+	// https://github.com/RedHatInsights/cloud-connector/blob/master/internal/pendo_transmitter/pendo_reporter.go#L51
+	reqBody, err := json.Marshal(track)
+	if err != nil {
+		logger.Error("marshaling the track information")
+		return err
+	}
+
+	// Prepare the request
+	url := c.Config.Clients.PendoBaseURL + "/track"
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(reqBody))
+	if err != nil {
+		logger.Error("creating request to track pendo event")
+		return err
+	}
+
+	// Add headers to the request
+	req.Header.Set("content-type", "application/json")
+	req.Header.Set(header.HeaderXPendoIntegrationKey, c.Config.Clients.PendoAPIKey)
+
+	// Launch request
+	resp, err := c.Client.Do(req)
+	switch {
+	case err != nil:
+		logger.Error("doing request to pendo service")
+		return err
+	case resp.StatusCode != http.StatusOK:
+		logger.Error("pendo service returned unsuccessful")
+		return fmt.Errorf("Pendo SendTrackEvent request unsuccessful: status: %v", resp.StatusCode)
+	}
+
+	logger.Debug("pendo metric track successfully")
+	return nil
+}
+
+//
+// ----- Private methods ------
+//
+
+func newClient(cfg *config.Config) *pendoClient {
+	if cfg == nil {
+		panic("'cfg' is nil")
+	}
+	if cfg.Clients.PendoBaseURL == "" {
+		panic("'PendoBaseURL' is empty")
+	}
+	if cfg.Clients.PendoAPIKey == "" {
+		panic("'PendoAPIKey' is empty")
+	}
+	if cfg.Clients.PendoRequestTimeoutSecs == 0 {
+		cfg.Clients.PendoRequestTimeoutSecs = 3
+	}
+	client := &http.Client{
+		Timeout: time.Duration(cfg.Clients.PendoRequestTimeoutSecs),
+	}
+	return &pendoClient{
+		Config: cfg,
+		Client: client,
+	}
+}
+
+func (c *pendoClient) guardSetMetadata(kind pendo.Kind, group pendo.Group, metrics pendo.SetMetadataRequest) error {
+	if kind == "" {
+		return fmt.Errorf("'kind' is an empty string")
+	}
+	if group == "" {
+		return fmt.Errorf("'group' is an empty string")
+	}
+	if metrics == nil {
+		return fmt.Errorf("'metrics' is nil")
+	}
+	for i := range metrics {
+		if metrics[i].VisitorID == "" {
+			return fmt.Errorf("'metrics[%d].VisitorID' is an empty string", i)
+		}
+	}
+	// TODO check 'kind' and 'group' contains valid characters
+	//      we need additional information to do that
+	return nil
+}
+
+func (c *pendoClient) guardSendTrackEvent(track *pendo.TrackRequest) error {
+	if track == nil {
+		return fmt.Errorf("'track' is nil")
+	}
+	if track.AccountID == "" {
+		return fmt.Errorf("'track.AccountID' is an empty string")
+	}
+	if track.Type != "track" {
+		return fmt.Errorf("'track.Type' is '%s', but '%s' was expected", track.Type, "track")
+	}
+	if track.Event == "" {
+		return fmt.Errorf("'track.Event' is an empty string")
+	}
+	if track.VisitorID == "" {
+		return fmt.Errorf("'track.VisitorID' is an empty string")
+	}
+	// It is strongly recommended
+	// if track.AccountID != "" {
```

Comment:
Thanks! it is commented because looking at the api I didn't see clear at the beginning if it is required or not: https://engageapi.pendo.io/?bash%23#e45be48e-e01f-4f0a-acaa-73ef6851c4ac 

Reading it again, my interpretation is the field is optional, so the verification does not make sense and the commented code can be removed.

Updating! :+1: